eqcorrscan/utils/plotting.py

```diff
@@ -2184,6 +2184,122 @@ def subspace_fc_plot(detector, stachans, **kwargs):
+        the origin time of each template in all detections.
+        Something like 2014-10-18T04:02:02
+        '''
+    import matplotlib.pyplot as plt
+    day = UTCDateTime(detections[0].detect_time.date)
+    times = [day.datetime, (day+(3600*24)).datetime]
+    plt.figure()
+    plt.plot(times, [0, 0])
+    if method == 'simple':
+        xdetects = [detect.detect_time.datetime for detect in detections]
+        ydetects = np.zeros(len(xdetects))
+        plt.plot(xdetects, ydetects, 'rx', markersize=7, label='child events')
+        try:
+            parents = [detection.template_name for detection in detections]
+            parents = list(set(parents))
+            number_parents = len(parents)
+            parents = [UTCDateTime(parent).datetime for parent in parents]
+            xparents = [parent for parent in parents if
+                        parent.date() == day.date]
+            yparents = np.zeros(len(parents))
+            plt.plot(xparents, yparents, '.', color='blue', label='parents')
+        except Exception as exception:
+            print(exception)
+            print(txt)
+    elif method == 'advance':
+        families = {}
+        for detect in detections:
+            template_name = detect.template_name
+            detect_time = detect.detect_time.datetime
+            if template_name in families.keys():
+                families[template_name].append(detect_time)
+            else:
+                families[template_name] = []
+                families[template_name].append(detect_time)
+        number_parents = len(families)
+        for family in sorted(families.keys()):
+            xdetects = families[family]
+            ydetects = np.zeros(len(xdetects))
+            plt.plot(xdetects, ydetects, 'x', markersize=7,
+                     label='{}({})'.format(family, len(xdetects)))
+        try:
+            xparents = [UTCDateTime(parent).datetime
+                        for parent in families.keys()
+                        if UTCDateTime(parent).date == day.date]
+            yparents = np.zeros(len(xparents))
+            plt.plot(xparents, yparents, '.', color='blue', markersize=7,
+                     label='parents')
+            plt.vlines(xparents, 0, 0.25, linewidth=0.6, alpha=0.5)
+            shift = dt.timedelta(seconds=700)
+            for parent in xparents:
+                plt.text(parent-shift, 0.1, parent, rotation=90)
+        except Exception as exception:
+            print(exception)
+            print(txt)
+    if number_parents > 30:
+        ncol = 2
+    else:
+        ncol = 1
+    legend = plt.legend(loc=(1.04, 0), edgecolor='g', ncol=ncol,
+                        title='parents({})\nchild-events({})'
+                        .format(number_parents, len(detections)))
+    legend._legend_box.align = "left"
+    plt.xlabel('Origin time')
+    plt.gca().axes.get_yaxis().set_visible(False)
+    fig = plt.gcf()
```

Comment:
Can you re-write this so that a user can pass a figure  as an argument please.

Reply:
I don't understand what kind of change you would like. can you explain more?